CorpusCallosum/segmentation/segmentation_inference.py

```diff
@@ -0,0 +1,279 @@
+import numpy as np
+import torch
+from monai import transforms
+
+from CorpusCallosum.transforms.segmentation_transforms import CropAroundACPC
+from CorpusCallosum.utils.checkpoint import YAML_DEFAULT as CC_YAML
+from FastSurferCNN.download_checkpoints import load_checkpoint_config_defaults
+from FastSurferCNN.download_checkpoints import main as download_checkpoints
+from FastSurferCNN.models.networks import FastSurferVINN
+
+
+def load_model(checkpoint_path, device=None):
+    """
+    Load the trained model from checkpoint
+    
+    Args:
+        checkpoint_path: Path to model checkpoint
+        device: torch device to load model to (defaults to CUDA if available)
+    
+    Returns:
+        model: Loaded model
+    """
+    if device is None:
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    
+
+    params = {
+        "num_classes": 3,
+        "num_filters": 71,
+        "num_filters_interpol": 32,
+        "num_channels": 9,
+        "kernel_h": 3,
+        "kernel_w": 3,
+        "kernel_c": 1,
+        "stride_conv": 1,
+        "stride_pool": 2,
+        "pool": 2,
+        "height": 128,
+        "width": 128,
+        "base_res": 1.0,
+        "interpolation_mode": "bilinear",
+        "crop_position": "top_left",
+        "out_tensor_width": 320,
+        "out_tensor_height": 320,
+    }
+    model = FastSurferVINN(params)
+    
+    download_checkpoints(cc=True)
+    cc_config = load_checkpoint_config_defaults(
+                "checkpoint",
+                filename=CC_YAML,
+            )
+    checkpoint_path = cc_config['segmentation']
+    
+    #model = torch.load(checkpoint_path, map_location=device, weights_only=False)
+    weights = torch.load(checkpoint_path, weights_only=True, map_location=device)
+    model.load_state_dict(weights)
+    model.eval()
+    model.to(device)
+    return model
+
+def run_inference(model, image_slice, AC_center, PC_center, voxel_size, device=None, transform=None):
+    """
+    Run inference on a single image slice
+    
+    Args:
+        model: Trained model
+        image_slice: Input image as numpy array
+        device: torch device to run inference on
+        transform: Optional custom transform pipeline
+    
+    Returns:
+        dict containing:
+            segmentation: Segmentation map if model produces segmentation
+            landmarks: Predicted landmarks if model produces localization
+    """
+    orig_shape = image_slice.shape
+    
+    if device is None:
+        #device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        device = next(model.parameters()).device
+        
+    def crop_around_acpc(img, ac, pc, vox_size):
+        return CropAroundACPC(keys=['image'], padding_mm=35, random_translate=0)(
+            {'image': img, 'AC_center': ac, 'PC_center': pc, 'res': vox_size}
+        )
+
+    # Preprocess slice
+    inputs = torch.from_numpy(image_slice[:,None,:256,:256]) # artifact from training script
```

Comment:
why is this slice always width & height 256, this is FastsurferVINN, should the midslice size not depend on the voxel size?
and if we go a bit lower you do another cropping step after anyways. Should this just be
```suggestion
    inputs = torch.from_numpy(image_slice[:, None]) # artifact from training script
``